ikarus/finiteelements/febases.hh

```diff
@@ -0,0 +1,180 @@
+      for (size_t i = 0; i < fe.size(); ++i)
+        globalIndices.push_back(localView_.index((localView_.tree().child(childIndex).localIndex(i))));
+    } else {
+      const auto& fe = localView_.tree().finiteElement();
+      for (size_t i = 0; i < fe.size(); ++i)
+        globalIndices.push_back(localView_.index(localView_.tree().localIndex(i)));
+    }
+  }
+
+  /**
+   * \brief A helper function to handle global indices of a power basis.
+   *
+   * \tparam ChildIndex Type of the index for the child, see dune/common/indices.hh
+   *
+   * \param globalIndices Output vector to store global indices.
+   * \param childIndex The index of the child.
+   */
+  template <typename ChildIndex>
+  void globalPowerFlatIndices(std::vector<GlobalIndex>& globalIndices, const ChildIndex childIndex) const {
+    if constexpr (requires { localView_.tree().child(childIndex).finiteElement(); }) {
+      const auto& fe             = localView_.tree().child(0).finiteElement();
+      constexpr int childrenSize = LocalView::Tree::degree();
+      for (size_t i = 0; i < fe.size(); ++i)
+        for (int j = 0; j < childrenSize; ++j)
+          globalIndices.push_back(localView_.index((localView_.tree().child(j).localIndex(i))));
+    } else {
+      const auto& fe             = localView_.tree().child(childIndex, 0).finiteElement();
+      constexpr int childrenSize = LocalView::Tree::template Child<childIndex>::Type::degree();
+      for (size_t i = 0; i < fe.size(); ++i)
+        for (int j = 0; j < childrenSize; ++j)
+          globalIndices.push_back(localView_.index((localView_.tree().child(childIndex, j).localIndex(i))));
+    }
+  }
+
+  /**
+   * \brief A helper function to handle global indices of a composite basis.
+   *
+   * \tparam ChildIndex Type of the index for the child, see dune/common/indices.hh
+   *
+   * \param globalIndices Output vector to store global indices.
+   * \param childIndex The index of the child.
+   */
+  template <typename ChildIndex>
+  void globalCompositeFlatIndices(std::vector<GlobalIndex>& globalIndices, const ChildIndex childIndex) const {
+    constexpr int localNumChildren = LocalView::Tree::template Child<childIndex>::Type::degree();
+    if constexpr (localNumChildren == 0)
+      globalScalarFlatIndices(globalIndices, childIndex);
+    else
+      globalPowerFlatIndices(globalIndices, childIndex);
+  }
+
+  void globalFlatIndicesImpl(std::vector<GlobalIndex>& globalIndices) const {
+    globalIndices.clear();
+    using namespace Dune::Indices;
+    if constexpr (isPower)
+      globalPowerFlatIndices(globalIndices, _0);
+    else if constexpr (isScalar)
+      globalScalarFlatIndices(globalIndices, _0);
+    else if constexpr (isComposite) {
+      Dune::Hybrid::forEach(Dune::Hybrid::integralRange(Dune::index_constant<numChildren>()), [&](const auto i) {
+        static_assert(not(LocalView::Tree::template Child<i>::Type::isComposite),
+                      "Composite Basis cannot have a composite basis");
```

Comment:
WE can call recursively again `globalCompositeFlatIndices(globalIndices, i);` here. Then it really works for all construction

Reply:
We cann talk about that tomorrow.